### PR TITLE
Ddcnl 5574

### PR DESCRIPTION
**Pull request by SandeepUmredkar**

### CODE REVIEW

File: app/views/interstitial/ViewNewsAndUpdatesView.scala.html [hunk 1]
```diff
@@ -17,38 +17,31 @@
...
+    @mainView(
+        pageTitle = messages("label.news_and_updates"),
+        showBackLink = true
+    ) {
 
-}
+        @h1("label.news_and_updates", elmId = Some("newsHeading"))
+        @for(newsAndContentModel <- newsAndContentModelList) {
+            <section class="app-markdown">
+                <h2 id="@newsAndContentModel.newsSectionName" autofocus>@newsAndContentModel.shortDescription</h2>
```
Comment (pascal22p) [inline app/views/interstitial/ViewNewsAndUpdatesView.scala.html:41]: This won't work. Only the selected news should have the autofocus attribute.

File: conf/application.conf [hunk 2]
```diff
@@ -354,6 +354,32 @@ feature {
...
+      short-description-cy: "1.25 percentage points uplift in National Insurance contributions (base64 encoded)"
+      content-cy: "<p>base64 encoded content with html</p>"
+      start-date: "2022-01-02"
+      end-date: "2023-01-02"
+    }
+    dynamicSection { // key used in url and anchor and also used to match the bit of code dealing with this
+      dynamic-content: true // indicate content needs to be picked up in the code itself
+      start-date: "2022-04-01"
+      end-date: "2022-06-01"
+    }
```
Comment (pascal22p) [inline conf/application.conf:379]: Delete

File: app/controllers/InterstitialController.scala [hunk 3]
```diff
@@ -153,9 +153,11 @@ class InterstitialController @Inject() (
     }
   }
 
-  def displayNewsAndUpdates: Action[AnyContent] = authenticate { implicit request =>
+  def displayNewsAndUpdates(lang: String): Action[AnyContent] = authenticate { implicit request =>
```
Comment (pascal22p) [inline app/controllers/InterstitialController.scala:156]: I don't think you need lang here. If you extend i18n in the class definition you should be able to extract the current lang.

Or even better do this in the newsAndTilesConfig. See here: https://github.com/hmrc/taxcalc-frontend/blob/d079c1ffee5236df02497ba792cab26ce8032f6b/app/config/ApplicationConfig.scala#L44

Comment (SandeepUmredkar) [reply app/controllers/InterstitialController.scala:156]: Will try this.

File: conf/application.conf (hunk 2 above)
Comment (SandeepUmredkar) [reply conf/application.conf]: Done

File: app/views/interstitial/ViewNewsAndUpdatesView.scala.html (hunk 1 above)
Comment (SandeepUmredkar) [reply app/views/interstitial/ViewNewsAndUpdatesView.scala.html]: Done

**Comment by SandeepUmredkar** [comment]
@pascal22p , I have resolved the code review comments and also corrected the autofocus behaivour

File: conf/application.conf [hunk 4]
```diff
@@ -354,6 +355,19 @@ feature {
   }
 }
 
+feature {
+  news {
+    nicSection {
+      short-description-en: "1.25 percentage points uplift in National Insurance contributions (base64 encoded)"
+      content-en: "<p>base64 encoded content with html</p>"
```
Comment (Nj147) [inline conf/application.conf:362]: Similar to the H2s class="govuk-body" is needed in these P tags for styling

Comment (Nj147) [inline conf/application.conf:362]: Should the content not match what is already there? Or is it going to be replaced by a different app-config

**Review by Nj147 — CHANGES_REQUESTED** [review]

File: app/views/interstitial/ViewNewsAndUpdatesView.scala.html [hunk 5]
```diff
@@ -17,38 +17,35 @@
...
+    ) {
 
-}
+        @h1("label.news_and_updates", elmId = Some("newsHeading"))
+        @for(newsAndContentModel <- newsAndContentModelList) {
+            <section class="app-markdown">
+                @if(selectedNewsSectionId.equals(newsAndContentModel.newsSectionName)) {
+                    <h2 id="@newsAndContentModel.newsSectionName" autofocus>@newsAndContentModel.shortDescription</h2>
+                } else {
+                    <h2 id="@newsAndContentModel.newsSectionName">@newsAndContentModel.shortDescription</h2>
```
[the remark this answers is not in the document]
Comment (SandeepUmredkar) [reply app/views/interstitial/ViewNewsAndUpdatesView.scala.html]: Cool, will add it.

**Review by Nj147 — DISMISSED** [review]

**Review by Luka-Abey — DISMISSED** [review]

**Review by Sachin300793 — APPROVED** [review]

**Pull request merged**